src/benchmarklib/tpcc/tpcc_table_generator.cpp

```diff
@@ -520,12 +540,30 @@ TpccTableGeneratorFunctions TpccTableGenerator::tpcc_table_generator_functions()
   return generators;
 }
 
-std::shared_ptr<Table> TpccTableGenerator::generate_tpcc_table(const std::string& tablename) {
-  auto generators = TpccTableGenerator::tpcc_table_generator_functions();
-  if (generators.find(tablename) == generators.end()) {
+std::shared_ptr<Table> TpccTableGenerator::generate_table(const std::string& tablename) {
+  if (tablename == "ITEM") {
+    return generate_items_table();
+  } else if (tablename == "WAREHOUSE") {
+    return generate_warehouse_table();
+  } else if (tablename == "STOCK") {
+    return generate_stock_table();
+  } else if (tablename == "DISTRICT") {
+    return generate_district_table();
+  } else if (tablename == "CUSTOMER") {
+    return generate_customer_table();
+  } else if (tablename == "HISTORY") {
+    return generate_history_table();
+  } else if (tablename == "ORDER") {
+    return generate_new_order_table();
+  } else if (tablename == "NEW_ORDER") {
+    auto order_line_counts = generate_order_line_counts();
+    return generate_order_table(order_line_counts);
+  } else if (tablename == "ORDER_LINE") {
+    auto order_line_counts = generate_order_line_counts();
```

Comment:
Will `NEW_ORDER` and `ORDER_LINE` be consistent if they call `generate_order_line_counts` twice?